### PR TITLE
KFLUXUI-1075 - add "Test result" column to regular PipelineRun list

### PR DESCRIPTION
**Pull request by rrosatti**
## Fixes 

Fixes https://issues.redhat.com/browse/KFLUXUI-1075. Not entirely, but one of the issues :)

## Description

In this PR we're handling one of the issues pointed out in KFLUXUI-1075. Here we're adding the `"Test result"` column into the regular PipelineRun list.

The main logic around the value for `"Test result"` is:

1. if the PLR itself has a test output result (`TEST_OUTPUT`), use it
2. If not, fetch the PLR’s TaskRuns and use their test output results. When multiple task runs have `TEST_OUTPUT`, the hook returns the "worst result" (e.g. `ERROR` over `FAILURE` over `SUCCESS`).

TaskRuns are only fetched when the PLR has no `TEST_OUTPUT` at pipeline level.

## Type of change

- [ ] Feature
- [ ] Bugfix
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Screen shots / Gifs for design review 

https://github.com/user-attachments/assets/f75f2635-9d19-4708-bf1f-5f63763b04f5

## How to test or reproduce?

1. navigate to Activity -> Pipeline runs tab
2. click "Manage columns" button and select "Test result"
3. the `Test result` column should be shown
4. :coffee:

I have prepared some examples of PLRs with different "test results" in case you wanna check it out: https://localhost:8080/ns/rh-ee-rgalvao-tenant/applications/testing-123/activity/pipelineruns?name=testing-test

## Browser conformance: 

- [ ] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

## Summary by CodeRabbit

* **New Features**
  * "Test result" column added and shown by default; dedicated cell shows spinner while loading and status icon/text when available, with namespace-aware loading and automatic pagination.

* **Bug Fixes / Improvements**
  * Tooltip clarified: pipeline-level TEST_OUTPUT preferred, otherwise falls back to task-run results with worst-case resolution. Column enabled in vulnerability view.

* **Tests**
  * Test suites run with a QueryClient context and expanded to cover mappings, edge cases, loading, pagination, and namespace behavior.

### CODE REVIEW

File: src/hooks/usePipelineRunTesOutputResult.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  ERROR: 4,
+};
+
+const usePipelineRunTesOutputResult = (
+  plr?: PipelineRunKind,
+  namespace?: string,
+): UsePipelineRunTesOutputResult => {
+  const [taskRuns, loaded, error, getNextPage, nextPageProps] = useTaskRunsForPipelineRuns(
+    namespace,
+    plr?.metadata?.name ?? '',
```
Comment (sahil143) [inline src/hooks/usePipelineRunTesOutputResult.ts]: This is bad, this will end up fetching all the taskrun in namespace

```suggestion
    plr.metadata.name,
```

File: src/hooks/usePipelineRunTesOutputResult.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    if (testStatus.result === 'WARNING') return runStatus.TestWarning;
+    if (testStatus.result === 'SKIPPED') return runStatus.Skipped;
+
+    return runStatus.Unknown;
+  }, [testStatus?.result]);
+
+  // when namespace is undefined we are not fetching task runs (e.g. PLR has TEST_OUTPUT); treat as not loading
+  const isLoading = namespace !== undefined ? !loaded || nextPageProps.isFetchingNextPage : false;
+
+  return [isLoading, convertedStatus, testStatus?.note];
```
Comment (sahil143) [inline src/hooks/usePipelineRunTesOutputResult.ts]:
```suggestion
  return [convertedStatus, isLoading, testStatus?.note];
```

**Comment by rrosatti** [comment]
@CodeRabbit full review

**Comment by rrosatti** [comment]
@CodeRabbit full review

**Comment by rrosatti** [comment]
> Tested locally, i can see the test result column is shown as default. However, when i click the help info of it, the content shows different font style. Is it my fake feeling?
> 
> Image
> BTW, please move the [WIP] from the PR title, it is quite confusing whether it is ready to review. :-)

About the "different font style", for me it looks the same 🤔 

And regarding the `[WIP]` in the PR title, thanks for pointing it out! It got unnoticed by me :D
